src/backup.rs

```diff
@@ -26,22 +29,52 @@ struct Backup {
 
 
 impl BackupOptions {
-    pub fn backup(&self, archive_path: &Path, source: &Path, report: &Report) -> Result<()> {
+    pub fn backup(&self, archive_path: &Path, source: &Path, report: &Report, excludes: Option<Vec<&str>>) -> Result<()> {
         let archive = try!(Archive::open(archive_path, &report));
         let band = try!(archive.create_band(&report));
+        let excludes = BackupOptions::get_excludes(excludes)?;
+
         let mut backup = Backup {
             block_dir: band.block_dir(),
             index_builder: band.index_builder(),
             report: report.clone(),
         };
+
         let source_iter = try!(sources::iter(source, report));
         for entry in source_iter {
-            try!(backup.store_one_source_entry(&try!(entry)));
+            let entry = entry?.clone();
+            let path = entry
+                .apath
+                .to_string()
+                .clone()
+                .chars()
+                .skip(1)
+                .collect::<String>();  // remove leading '/‘ from path
+
+            if excludes.is_some() && excludes.clone().unwrap().matches(&path).len() > 0 {
+                info!("Skipping {}", path);
+                continue;
+            }
+            try!(backup.store_one_source_entry(&entry));
         }
         try!(backup.index_builder.finish_hunk(report));
         try!(band.close(&backup.report));
         Ok(())
     }
+
+
+    fn get_excludes(excludes: Option<Vec<&str>>) -> Result<Option<GlobSet>> {
```

Comment:
This isn't backup-specific and should be somewhere else, maybe in a new `excludes.rs`. Also, it can then have unit tests there.